exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/vendors/H2.kt

```diff
@@ -43,7 +43,7 @@ internal object H2FunctionProvider : FunctionProvider() {
     ): String {
         val uniqueIdxCols = table.indices.filter { it.unique }.flatMap { it.columns.toList() }
         val primaryKeys = table.primaryKey?.columns?.toList() ?: emptyList()
-        val uniqueCols = columns.filter { it in uniqueIdxCols } + primaryKeys
+        val uniqueCols = (columns.filter { it in uniqueIdxCols } + primaryKeys).distinct()
```

Comment:
I still think that `columns.filter { it in uniqueIdxCols }` is always the same as uniqueIdxCols as it's not possible to create an index with columns from another table. 
```suggestion
        val uniqueCols = (uniqueIdxCols  + primaryKeys).distinct()
```